swarmrl/losses/policy_gradient_loss.py

```diff
@@ -44,140 +44,93 @@ def __init__(self, value_function: ExpectedReturns = ExpectedReturns()):
-                is preserved in the model training.
-        actor : Network
-                Actor model to use in the analysis.
-        critic : Network
-                Critic model to use in the analysis.
+        network : FlaxModel
+            The actor-critic network that approximates the policy.
+        network_params : FrozenDict
+            Parameters of the actor-critic model used.
+        feature_data : np.ndarray (n_time_steps, n_particles, feature_dimension)
+            Observable data for each time step and particle within the episode.
+        action_indices : np.ndarray (n_time_steps, n_particles)
+            The actions taken by the policy for all time steps and particles during one
+            episode.
+        rewards : np.ndarray (n_time_steps, n_particles)
+            The rewards received for all time steps and particles during one episode.
+
 
         Returns
         -------
         loss : float
-                The loss for the episode.
+            The loss of the actor-critic network for the last episode.
         """
-        actor_apply_fn = jax.vmap(actor.apply_fn, in_axes=(None, 0))
+
         # (n_timesteps, n_particles, n_possibilities)
-        logits = actor_apply_fn({"params": actor_params}, feature_data)
+        logits, predicted_values = network(network_params, feature_data)
+        predicted_values = predicted_values.squeeze()
         probabilities = jax.nn.softmax(logits)  # get probabilities
         chosen_probabilities = gather_n_dim_indices(probabilities, action_indices)
-        log_probs = np.log(chosen_probabilities + 1e-8)
+        log_probs = jnp.log(chosen_probabilities + 1e-8)
         logger.debug(f"{log_probs.shape=}")
 
-        value_function_values = self.value_function(rewards)
-        logger.debug(f"{value_function_values.shape}")
+        returns = self.value_function(rewards)
+        logger.debug(f"{returns.shape}")
 
-        critic_apply_fn = jax.vmap(critic.__call__, in_axes=(0))
-        critic_values = critic_apply_fn(feature_data)[
-            :, :, 0
-        ]  # zero for trivial dimension
-        logger.debug(f"{critic_values.shape=}")
+        logger.debug(f"{predicted_values.shape=}")
 
         # (n_timesteps, n_particles)
-        advantage = value_function_values - critic_values
+        advantage = returns - predicted_values
         logger.debug(f"{advantage=}")
 
-        loss = -1 * ((log_probs * advantage).sum(axis=0)).mean()
-        logger.debug(f"{loss=}")
+        actor_loss = -1 * ((log_probs * advantage).sum(axis=0)).mean()
+        logger.debug(f"{actor_loss=}")
 
-        return loss
+        critic_loss = jnp.sum(optax.huber_loss(predicted_values, returns), axis=0)
 
-    def _compute_critic_loss(
-        self,
-        critic_params: FrozenDict,
-        feature_data: np.ndarray,
-        rewards: np.ndarray,
-        critic: Network,
-    ):
+        critic_loss = jnp.mean(critic_loss)
+
+        return actor_loss + 0.5 * critic_loss
```

Comment:
why the 0.5 scaling in the critic loss?

Reply:
Good point! 
Just copied that from the PPO algo
https://www.tensorflow.org/tutorials/reinforcement_learning/actor_critic
They use 1.0 for the scaling. 
Will change that.

Reply:
Cheers